src/components/Observations/Toolbar.js

```diff
@@ -1,53 +1,139 @@
+  const numUnuploadedObs = useNumUnuploadedObservations( );
+  const {
+    stopUpload,
+    uploadInProgress,
+    startUpload,
+    error: uploadError
+  } = useUploadObservations( allObsToUpload );
+
   const loading = obsEditContext?.loading;
   const syncObservations = obsEditContext?.syncObservations;
 
+  const totalObsToUpload = Math.max( allObsToUpload.length, numUnuploadedObs );
+  const progress = ( totalObsToUpload - numUnuploadedObs + 1 ) / ( totalObsToUpload + 1 );
+
+  const getSyncClick = ( ) => {
+    if ( numUnuploadedObs > 0 ) {
+      return startUpload;
+    }
+    return syncObservations;
+  };
+
+  const cleanProgress = ( progressToClean: number ) => {
+    if ( Number.isNaN( progressToClean ) || progressToClean < 0 ) {
+      return 0;
+    }
+
+    return progressToClean;
+  };
+
+  const getStatusText = ( ) => {
+    if ( !uploadInProgress && totalObsToUpload > 0 ) {
+      return t( "UPLOAD-X-OBSERVATIONS", { count: numUnuploadedObs } );
+    }
+    if ( totalObsToUpload > 0 ) {
+      return t( "Uploading-X-Observations", { count: numUnuploadedObs } );
+    }
+    return null;
+  };
+
+  const getSyncIconColor = ( ) => {
+    if ( uploadInProgress || totalObsToUpload > 0 ) {
+      return colors.inatGreen;
+    }
+    return colors.darkGray;
+  };
+
+  const statusText = getStatusText( );
+
+  /* eslint-disable react-native/no-inline-styles */
   return (
-    <View className="py-5 flex-row justify-between bg-white">
-      {currentUser ? (
-        <Pressable
-          onPress={syncObservations}
-          className="mx-3"
-          accessibilityRole="button"
-          disabled={loading}
-        >
-          <IconMaterial name="sync" size={30} />
-        </Pressable>
-      ) : (
-        <View className="mx-3" />
-      )}
-      {loading && <ActivityIndicator />}
-      <View className="flex-row mx-3">
-        <Pressable
-          onPress={( ) => setView( "list" )}
-          accessibilityRole="button"
-        >
-          <IconMaterial name="format-list-bulleted" size={30} />
-        </Pressable>
-        <Pressable
-          onPress={( ) => setView( "grid" )}
-          testID="ObsList.toggleGridView"
+    <View className="bg-white border-b border-[#e8e8e8]">
+      <View className="py-5 flex flex-row px-[15px]">
+        <View className="flex justify-between items-center flex-row grow">
+          <View className="flex items-center flex-row">
+            {currentUser && (
+              <TouchableOpacity className="mr-3" accessibilityRole="button">
+                <IconMaterial name="language" size={30} />
+              </TouchableOpacity>
+            )}
+            <TouchableOpacity
+              onPress={getSyncClick( )}
+              accessibilityRole="button"
+              disabled={loading || uploadInProgress}
+              accessibilityState={{ disabled: loading || uploadInProgress }}
+            >
+              <IconMaterial name="sync" size={26} color={getSyncIconColor( )} />
```

Comment:
Can we add the clockwise rotation animation when an upload is in progress? We'll swap out the icon later, but it would be nice to have the animation now.